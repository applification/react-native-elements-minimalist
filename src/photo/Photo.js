import React, { Component } from 'react';
import { Image } from 'react-primitives';
import PropTypes from 'prop-types';

class PhotoElement extends Component {
  static propTypes = {
    imageURL: PropTypes.string.isRequired,
  };

  render() {
    return (
      <Image
        style={{ width: 350, height: 350 }}
        source={{ uri: this.props.imageURL }}
      />
    );
  }
}

export default PhotoElement;
