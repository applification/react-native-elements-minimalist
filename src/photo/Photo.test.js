import React from 'react';
import { Dimensions } from 'react-native';
import renderer from 'react-test-renderer';

import Photo from './Photo';

describe('IMAGE', () => {
  xit('Renders image to full device width', () => {
    const component = renderer
      .create(
        <Photo
          source={{
            uri:
              'https://scontent-lht6-1.cdninstagram.com/t51.2885-15/sh0.08/e35/p640x640/26152588_1188914984576015_4446380707253583872_n.jpg',
          }}
        />
      )
      .toJSON();
    const width = component.props.style[0].width;
    const deviceWidth = Dimensions.get('window').width;

    expect(width).toBe(deviceWidth);
    expect(component).toMatchSnapshot();
  });
});
