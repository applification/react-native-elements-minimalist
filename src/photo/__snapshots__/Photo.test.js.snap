// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IMAGE Renders image 1`] = `
<Image
  source={
    Object {
      "uri": "https://scontent-lht6-1.cdninstagram.com/t51.2885-15/sh0.08/e35/p640x640/26152588_1188914984576015_4446380707253583872_n.jpg",
    }
  }
  style={
    Object {
      "height": 350,
      "width": 350,
    }
  }
/>
`;
